outlines/models/openai.py

```diff
@@ -111,36 +147,72 @@ async def generate_choice(
                 )
 
             tokenizer = tiktoken.encoding_for_model(model_name)
-            encoded: List[List[int]] = [tokenizer.encode(word) for word in is_in]
 
             decoded_samples = []
             for _ in range(samples):
+                is_in_left = is_in.copy()
                 decoded: List[str] = []
-                for i in range(max([len(word) for word in encoded])):
-                    mask = {}
-                    for word, tokenized_word in zip(is_in, encoded):
-                        if not word.startswith("".join(decoded)):
-                            continue
-                        try:
-                            mask[tokenized_word[i]] = 100
-                        except IndexError:
-                            pass
+
+                greedy = False  # we try to generate the full response at each iteration
```

Comment:
We might as well remove the greedy case altogether

Reply:
Say you have to tokenized choices `[T1, TX]`, `[T2, TX]`. You would unmask the three tokens and request `2` tokens from the API. If the system is very biased towards `TX` it may answer `[TX, TX]`, and  you have gained nothing. Repeating the same call you would enter a loop. Thus, it make sense then to try a _greedy_ step next just `T1` and `T2` unmasked, such that you know that you are guaranteed to move forward one token forward.

Reply:
Ok, this makes sense.